internal/cmd/controller/gitops/reconciler/gitjob_controller.go

```diff
@@ -197,10 +206,14 @@ func (r *GitJobReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 			if err == nil && commit != "" {
 				gitrepo.Status.Commit = commit
 			}
+			if err != nil {
+				r.Recorder.Event(gitrepo, fleetevent.Warning, "Failed", err.Error())
+			}
 		}
 
 		if gitrepo.Status.Commit != "" {
 			if err := r.validateExternalSecretExist(ctx, gitrepo); err != nil {
+				r.Recorder.Event(gitrepo, fleetevent.Warning, "Failed", err.Error())
```

Comment:
Which criteria were used to determine whether or not to generate events?
For instance, do we want to generate an event if failing to create git job resources or to delete a job?

Reply:
Criteria in this case is to create the event in the case that the information we give to the user means something he/she can do something about.

For example, in this case we're checking for the secrets that the user creates.
If something goes wrong.. the user is able to fix it.

Later in the same file... we call `createJobRBAC`, for example.
I didn't add any event there because if that fails... it's an internal error and the user is not going to be able to do anything about it.

For those cases I was thinking on adding a more generic error, maybe.